src/Initialization.asm

```diff
@@ -0,0 +1,205 @@
+section .bss
+   BackgroundBufferSegment resw 1 ; where is stored the dynamic background (dynamic cuz gums are disappearing). Used to restore the screen after a ghost's passage
+   ScreenBufferSegment resw 1
+
+   
+section .text
+
+   BuildScreenBuffer:
+      call heapInit
+      mov ax, (SCREEN_HEIGHT*SCREEN_WIDTH/16) ; in ax, the number of paragraph to allocate (1 para = 16bits)
+      call heapAllocParagraph
+      mov [ScreenBufferSegment],bx
+   
+   ret
+
+   UpdateScreen:
+      push ds
+      push es
+
+      mov ax,cs:[ScreenBufferSegment]
+      mov ds,ax ; backBuffer segment adress
+      mov ax,0xa000 ; video memory segment adress
+      mov es,ax 
+      xor si,si ; beginning of the backbuffer (0)
+      xor di,di ; beginning of the video memory (0)
+      mov cx, SCREEN_HEIGHT*SCREEN_WIDTH
+      rep movsb
+
+      pop es
+      pop ds
+   ret
+
+   
+
+   SetVideoMode:
+
+      mov ah, 00h ; chose the video mode setting function at the 10h interruption
+      mov al, 13h ; set video mode option to 320 x 200 256 colors
+      int 10h ; call the dos interupt regarding to ax
+
+      ; write the adress of video memory (A000) in the register which contains the address of the destination segment used by the automatic data transfer functions (like movs, stos, ...)
+      push word [ScreenBufferSegment] 
+      pop es
+
+   ret
+
+   ClearScreen:
+      ;clear the screen by filling it with a unique color (stored in al)
+      mov al, 0x29 ; color to fill the screen (white = 0x0F, black = 0x00)
+            
+      ;set the destination 'es:di' :
+      push word [ScreenBufferSegment] ; video memory adress = 0xA000
+      pop es ; define it as the adress of the destination segment
+      mov di, 0 ; begin the offset (pixel position) at 0
+      mov cx, SCREEN_WIDTH*SCREEN_HEIGHT ; how many times 'rep' action will be repeated
+      rep stosb ; store (byte per byte) the content of al into es:di, es = 0xA000, di increasing from 0 to 200*320
+   ret
+
+   BuildBackgroundBuffer: 
+      mov ax, (SCREEN_HEIGHT*SCREEN_WIDTH/16) ; in ax, the number of paragraph to allocate (1 para = 16bits)
```

Comment:
Since the background buffer is stored in tiles and not in pixels, shouldn't its size be divided by the area of a tile?

Reply:
Alright, I misunderstood your code. I will merge now but please add more documentation later on.